Kb new dev

Yuxuan-Lin opened the pull request:
Changes:
Account Settings (In progress: switching between senior and junior available)
Official Page (Finished)
Home (myOpps Fully Connected)
Oppoutuniteis(Details Debugged)
Messages(Search; Create New Conversation; Realtime all fully functional)
Minor renaming completed(billFunction => setIUpOppScreen)

Commit:
Account Settings HTML and CSS Completed!

Commit:
Official Page Completed

Commit:
Fixed loop between junior and senior side

Commit:
added upload resume button in account setting UI

Commit:
fixed error in oppView.renderOpps function

Yuxuan-Lin committed:
Merge branch 'master' into kb-new-dev

victorjzsun commented on src/js/models/opportunities.js:
```diff
@@ -6,12 +6,53 @@ export default class Opportunities{
 		try{
 			//const oppArr = {};
-			await db.collection('Opportunities').get().then(snapshot => {
+			await db.collection('NewOpportunities').get().then(snapshot => {
```
Why are you using NewOpportunities? It's not usable yet

victorjzsun replied:
Created issue to fix later

victorjzsun commented on src/js/controllers/homeController.js:
```diff
@@ -1,23 +1,38 @@
     
     //3) Render UI
     homeView.renderProfile(state.home.profile);
     homeView.renderAbout(state.home.profile);
     homeView.renderExps(state.home.profile.experience);
-    homeView.renderOpps(state.home.profile,state.opp.opps);
+    if (state.user.isSenior){
+        oppView.renderOpps(state.home.seniorOpps);
+        //console.log(state.home.seniorOpps[0].data());
+    } else {
+        homeView.renderOpps(state.home.myOpps,state.opp.opps);
+    }
+
```
Instead of making so many function calls, can we just call one function like `homeView.renderHome(...)` which does whatever it needs to render the whole page?

victorjzsun commented on src/js/index.js:
```diff
@@ -1,140 +1,221 @@
+    
+};
+
+const renderAccountSettings = async function(state){
```
Shouldn't we have an account settings Model and View as well? Again so we don't pile random stuff onto index.js

victorjzsun commented on src/js/models/opportunities.js:
```diff
@@ -6,12 +6,53 @@ export default class Opportunities{
+				uid: state.user.uid
+			})
+		} catch (error){
+			alert("??????????????" + error);
```
Please do not push useless code
```suggestion
			alert(error);

```

victorjzsun commented on src/js/index.js:
```diff
@@ -1,140 +1,221 @@
+                if(uploadBtn){
+                    console.log('test');
+                }
+            });
```
What is this?
```suggestion
```

Yuxuan-Lin replied:
This is the button created for Leo to test the storage functions in firebase.

victorjzsun replied:
Why is this on this branch and not on @haohao1331's branch then?

victorjzsun replied:
At the very least, you should name it appropriately, not `???????`
```suggestion

            //Testing button, to be removed later
            document.querySelector("#uploader").addEventListener('click', e => {
                e.preventDefault();
                const uploadBtn = e.target.closest('#uploader');

                if(uploadBtn){
                    console.log('test');
                }
            });
```

victorjzsun reviewed (changes requested):
I'll approve after comments are resolved

Yuxuan-Lin committed:
Update src/js/models/opportunities.js

Co-authored-by: Victor Sun <victorjsun@gmail.com>

Yuxuan-Lin committed:
Reverse Changes

Reversed front-end-dev changes in css. The front-end-dev changes broke UI and it won't be applied until tested.

rg3515 reviewed (dismissed):
lgtm

rg3515 committed:
Update src/js/index.js change comment ??? to proper comment

Co-authored-by: Victor Sun <victorjsun@gmail.com>

victorjzsun reviewed (approved):
lgtm
